src/relay/src/RelayLogger.js

```diff
@@ -0,0 +1,78 @@
+// @flow
+
+/* eslint-disable no-console */
+
+import { isBrowser } from '@adeira/js';
+import type { LogEvent, OperationAvailability } from 'relay-runtime';
+
+function renderQueryAvailability(
+  queryAvailability: OperationAvailability,
+  shouldFetch: boolean,
+): string {
+  if (queryAvailability.status === 'available') {
+    const fetchTime = queryAvailability.fetchTime;
+    if (fetchTime != null) {
+      const seconds = Math.round((Date.now() - fetchTime) / 1000);
+      return `🟢 ${seconds}s old${shouldFetch ? ', re-fetching' : ''}`;
+    }
+    return `🟢 ${shouldFetch ? 're-fetching' : ''}`;
+  } else if (queryAvailability.status === 'stale') {
+    return `🟠 ${shouldFetch ? 're-fetching' : ''}`;
+  } else if (queryAvailability.status === 'missing') {
+    return `🔴 ${shouldFetch ? 'fetching' : ''}`;
+  }
+  (queryAvailability.status: empty);
```

Comment:
All clear, I just don't get this line

Reply:
It is an exhaustive check: https://mrtnzlml.com/docs/flow/patterns/exhaustive-checking

Basically, in case I'd forget to implement one of the statuses (or a new one will be added in the future) it will throw the following error:

<img width="732" alt="Screen Shot 2021-09-06 at 8 46 17" src="https://user-images.githubusercontent.com/978368/132226853-da91de3d-9181-4092-a4a2-755511b20e5e.png">

In other words, it says: the status was not completely exhausted and `stale` needs to be implemented.